src/lib/syncable_ledger/test_mask.ml

```diff
@@ -0,0 +1,108 @@
+open Core
+
+(* Input for testing the interaction between merkle *)
+
+module Make (Input : sig
+  val depth : int
+
+  val mask_layers : int
+end) =
+struct
+  open Merkle_ledger_tests.Test_stubs
+
+  module Hash = struct
+    type t = Hash.t [@@deriving sexp, hash, compare, bin_io, eq]
+
+    type account = Account.t
+
+    let merge = Hash.merge
+
+    let hash_account = Hash.hash_account
+
+    let to_hash = Fn.id
+
+    let empty_account = hash_account Account.empty
+  end
+
+  module Maskable_and_mask =
+    Merkle_ledger_tests.Test_mask.Make_maskable_and_mask_with_depth (Input)
+
+  module L = struct
+    open Merkle_ledger_tests.Test_stubs
+    module Base_db = Maskable_and_mask.Base_db
+    module Any_base = Maskable_and_mask.Any_base
+    module Base = Any_base.M
+    module Mask = Maskable_and_mask.Mask
+    module Maskable = Maskable_and_mask.Maskable
+    include Mask.Attached
+
+    (* Each account for a layer of a mask will all have the same balance.
+    Specifically, the top most layer will have a balance of `balance` and
```

Comment:
a bit confusing here, not sure how "top most" and "afterward" are related.

Does this mean: 

  All accounts in the first mask added, which is at layer 0, has the balance given by `balance`. For a mask at layer `n` with a balance `b`, a mask added at layer `n + 1`, accounts have balance `2 * b`.